group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1242,7 +1304,11 @@ public void scheduleLoadOperation(
                         case FAILED:
                         case INITIAL:
                             context.transitionTo(CoordinatorState.LOADING);
-                            loader.load(tp, context.coordinator).whenComplete((state, exception) -> {
+                            loader.load(tp, context.coordinator).whenComplete((summary, exception) -> {
+                                if (summary != null) {
+                                    log.debug("Finished loading tp={}, LoadSummary={}", tp, summary);
+                                    runtimeMetrics.partitionLoadSensor.record(summary.loadTimeMs(), summary.endTimeMs(), false);
```

Comment:
* We should do this in the `scheduleInternalOperation` below. Could we also log the time and the other information in the final log message? That would be super useful.
* Accessing `partitionLoadSensor` is not safe in the current implementation. I think that we should rather define an method in the interface to hide it.

Reply:
1) ack.
2) the record method is synchronized so it is still safe, no?

Reply:
2. My point was more about the fact that partitionLoadSensor may or may not be initialized based on the current organization of the code so this could be a source of NPE.